src/components/DownloadButton/DownloadButton.tsx

```diff
@@ -1,23 +1,152 @@
-import React from "react"
-import { DownloadButtonProps } from "./DownloadButton.types"
+import React, { useCallback, useMemo } from "react"
+import { useAdvancedUserAgentData } from "@dcl/hooks"
+import Download from "@mui/icons-material/Download"
+import { config } from "../../config"
+import { CDNSource, getCDNRelease } from "../../modules/cdnReleases"
+import { triggerFileDownload } from "../../modules/file"
 import {
+  addQueryParamsToUrlString,
+  updateUrlWithLastValue,
+} from "../../modules/url"
+import { setUserAgentArchitectureDefaultByOs } from "../../modules/userAgent"
+import {
+  DownloadButtonProps,
+  DownloadOption,
+  OperativeSystem,
+} from "./DownloadButton.types"
+import {
+  DownloadButtonAppleIcon,
   DownloadButtonLabelContainer,
   DownloadButtonStyled,
+  DownloadButtonWindowsIcon,
 } from "./DownloadButton.styled"
 
 const DownloadButton = React.memo((props: DownloadButtonProps) => {
-  const { href, onClick, label, startIcon, endIcon } = props
+  const {
+    onClick,
+    label = "Download",
+    trackingId,
+    href,
+    startIcon,
+    endIcon,
+  } = props
+
+  const [isLoadingUserAgentData, userAgentData] = useAdvancedUserAgentData()
+
+  // Handle OS detection from URL params
+  const windowSearchParams =
+    typeof window !== "undefined"
+      ? new URLSearchParams(window.location.search)
+      : undefined
+  const searchParams = new URLSearchParams(windowSearchParams)
+  const os = searchParams.get("os")
+
+  if (userAgentData && os) {
+    setUserAgentArchitectureDefaultByOs(userAgentData, os as OperativeSystem)
+  }
```

Comment:
What if we put this inside a useEffect?

```suggestion
useEffect(() => {
  if (userAgentData && os) {
    setUserAgentArchitectureDefaultByOs(userAgentData, os as OperativeSystem)
  }
}, [ ])
```